### PR TITLE
Add handling for de:"Bienenwachs, Weiß und Gelb"

### PR DESCRIPTION
**Pull request by jolesh**
try 1 can't be as simple as that, right?

### CODE REVIEW

File: lib/ProductOpener/Ingredients.pm [hunk 1]
```diff
@@ -3757,7 +3757,11 @@ sub preparse_ingredients_text($$) {
 
 	$log->debug("preparse_ingredients_text - before language specific preparsing", { text => $text }) if $log->is_debug();
 
-	if ($product_lc eq 'es') {
+	if ($product_lc eq 'de') {
+		# deletes comma in "Bienenwachs, weiß und gelb" since it is just one ingredient
+		$text =~ s/Bienenwachs, weiß und gelb/Bienenwachs weiß und gelb /ig;
```
Comment (stephanegigandet) [inline lib/ProductOpener/Ingredients.pm:3762]: It should work. It will insert an extra space after "gelb" though.

Comment (jolesh) [reply lib/ProductOpener/Ingredients.pm:3762]: Okay fix is on the way

**Comment by jolesh** [comment]
@stephanegigandet I think I have to change the test to sth more simple.

**Comment by stephanegigandet** [comment]
Here is the output of the tests:

`````
#   Failed test at ../t/ingredients_parsing.t line 308.
#          got: 'zucker, glukosesirup, glukose-fruktose-sirup, st�rke, 8,5% s��holzsaft, brauner zuckersirup, modifizierte st�rke, aromen, pflanzliches �l (sonnenblume), �berzugsmittel: bienenwachs wei� und gelb'
#     expected: ''
Original ingredients: Zucker, Glukosesirup, Glukose-Fruktose-Sirup, Stärke, 8,5% Süßholzsaft, brauner Zuckersirup, modifizierte Stärke, Aromen, pflanzliches Öl (Sonnenblume), Überzugsmittel: Bienenwachs, weiß und gelb (de)
Ingredients (de): Zucker, Glukosesirup, Glukose-Fruktose-Sirup, Stärke, 8,5% Süßholzsaft, brauner Zuckersirup, modifizierte Stärke, Aromen, pflanzliches Öl (Sonnenblume), Überzugsmittel: Bienenwachs (weiß und gelb)
Preparsed: Zucker, Glukosesirup, Glukose-Fruktose-Sirup, Stärke, 8,5% Süßholzsaft, brauner Zuckersirup, modifizierte Stärke, Aromen, pflanzliches Öl (Sonnenblume), Überzugsmittel: Bienenwachs (weiß und gelb)
Use of uninitialized value $expected in lc at ../t/ingredients_parsing.t line 308.
not ok 117
#   Failed test at ../t/ingredients_parsing.t line 308.
#          got: 'zucker, glukosesirup, glukose-fruktose-sirup, st�rke, 8,5% s��holzsaft, brauner zuckersirup, modifizierte st�rke, aromen, pflanzliches �l (sonnenblume), �berzugsmittel: bienenwachs (wei� und gelb)'
#     expected: ''

The first one shows that the ingredients are corrected as we wanted (removed the comma), the second one shows that the parenthesis were not removed.

I'll add comments explaining why.

File: lib/ProductOpener/Ingredients.pm [hunk 2]
```diff
@@ -3757,7 +3757,13 @@ sub preparse_ingredients_text($$) {
 
 	$log->debug("preparse_ingredients_text - before language specific preparsing", { text => $text }) if $log->is_debug();
 
-	if ($product_lc eq 'es') {
+	if ($product_lc eq 'de') {
+		# deletes comma in "Bienenwachs, weiß und gelb" since it is just one ingredient
+		$text =~ s/Bienenwachs, weiß und gelb/Bienenwachs weiß und gelb/ig;
+		# deletes brackets in "Bienenwachs, weiß und gelb" since it is just one ingredient
+		$text =~ s/Bienenwachs (weiß und gelb)/Bienenwachs weiß und gelb/ig;
```
Comment (stephanegigandet) [inline lib/ProductOpener/Ingredients.pm:3764]: The parenthesis in regular expressions are used to group patterns. If you want to match parenthesis, they need to be escaped.

```suggestion
		$text =~ s/Bienenwachs \(weiß und gelb\)/Bienenwachs weiß und gelb/ig;
```

File: t/ingredients_parsing.t [hunk 3]
```diff
@@ -180,6 +180,8 @@ my @lists =(
 
 	["de","Wasser, Kohlensäure, Farbstoff Zuckerkulör E 150d, Süßungsmittel Aspartam* und Acesulfam-K, Säuerungsmittel Phosphorsäure und Citronensäure, Säureregulator Natriumcitrat, Aroma Koffein, Aroma. enthält eine Phenylalaninquelle", "Wasser, Kohlensäure, Farbstoff : Zuckerkulör e150d, Süßungsmittel : Aspartam* und Acesulfam-K, Säuerungsmittel : Phosphorsäure und Citronensäure, Säureregulator : Natriumcitrat, Aroma Koffein, Aroma. enthält eine Phenylalaninquelle"],
 	["de","Farbstoffe Betenrot, Paprikaextrakt, Kurkumin","farbstoffe : betenrot, paprikaextrakt, kurkumin"],
+	["de","Zucker, Glukosesirup, Glukose-Fruktose-Sirup, Stärke, 8,5% Süßholzsaft, brauner Zuckersirup, modifizierte Stärke, Aromen, pflanzliches Öl (Sonnenblume), Überzugsmittel: Bienenwachs, weiß und gelb"],
```
Comment (stephanegigandet) [inline t/ingredients_parsing.t:183]: The configuration for the ingredient preparsing tests has 3 arguments : ["de", "the original list of ingredients", "the transformed list of ingredients"]. Here it is missing the 3rd argument.

```suggestion
	["de","Zucker, Glukosesirup, Glukose-Fruktose-Sirup, Stärke, 8,5% Süßholzsaft, brauner Zuckersirup, modifizierte Stärke, Aromen, pflanzliches Öl (Sonnenblume), Überzugsmittel: Bienenwachs, weiß und gelb", Zucker, Glukosesirup, Glukose-Fruktose-Sirup, Stärke, 8,5% Süßholzsaft, brauner Zuckersirup, modifizierte Stärke, Aromen, pflanzliches Öl (Sonnenblume), Überzugsmittel: Bienenwachs weiß und gelb"],
```

Comment (jolesh) [reply t/ingredients_parsing.t:183]: Ah, okay thanks I was wondering how it knows what we expects.

**Review by stephanegigandet — APPROVED** [review]
Thank you!

**Pull request merged**